server/src/test/scala/ch.epfl.scala.index.server/ArtifactScalaVersionSupportTest.scala

```diff
@@ -0,0 +1,128 @@
+  ) {
+    ArtifactScalaVersionSupport(
+      Map(
+        `7.0.0` -> Seq(
+          ScalaJvm(`2.12`),
+          ScalaJvm(`2.13`),
+          ScalaJvm(`3.0.0-RC3`)
+        )
+      )
+    ).summaryOfLatestArtifactsSupportingScalaVersions should include(
+      "Scala 3.0.0-RC3, 2.13, 2.12"
+    )
+  }
+
+  it(
+    "should convey the latest artifact version available for each Scala language version"
+  ) {
+    val summary = ArtifactScalaVersionSupport(
+      Map(
+        `7.0.0` -> Seq(ScalaJvm(`2.11`)),
+        `7.1.0` -> Seq(ScalaJvm(`2.11`)),
+        `7.2.0` -> Seq(ScalaJvm(`2.12`)),
+        `7.3.0` -> Seq(ScalaJvm(`2.12`))
+      )
+    ).summaryOfLatestArtifactsSupportingScalaVersions
+
+    // these artifact versions are not the latest available support for any Scala language version, so uninteresting:
+    summary should not(include("7.0.0") or include("7.2.0"))
+
+    summary should include("7.1.0 (Scala 2.11)")
+    summary should include("7.3.0 (Scala 2.12)")
+  }
+
+  it(
+    "should, for brevity, not mention a Scala language version more than once, even if it occurs in multiple artifact versions being mentioned"
+  ) {
+    val summary = ArtifactScalaVersionSupport(
+      Map(
+        `7.1.0` -> Seq(ScalaJvm(`2.11`), ScalaJvm(`2.12`)),
+        `7.2.0` -> Seq(ScalaJvm(`2.12`))
+      )
+    ).summaryOfLatestArtifactsSupportingScalaVersions
+
+    // it happens that two artifact versions that support Scala 2.12 will be mentioned...
+    assert(summary.contains("7.1.0") && summary.contains("7.2.0"))
+
+    // ...but for brevity no Scala language version (eg Scala 2.12 in this case) should be mentioned more than once...
+    countMatches(summary, "2.12") shouldBe 1
+
+    // ...specifically it should be listed  against the *latest* artifact that supports that Scala language version
+    summary should include("7.2.0 (Scala 2.12)")
+  }
+
+  it(
+    "should, for brevity, only mention the *latest* Scala language versions available for any given Scala binary version family"
+  ) {
+    ArtifactScalaVersionSupport(
+      Map(
+        `7.0.0` -> Seq(ScalaJvm(`2.13`), ScalaJvm(`3.0.0-M3`)),
+        `7.1.0` -> Seq(ScalaJvm(`2.13`), ScalaJvm(`3.0.0-RC2`)),
+        `7.2.0` -> Seq(ScalaJvm(`2.13`), ScalaJvm(`3.0.0-RC3`))
+      )
+    ).summaryOfLatestArtifactsSupportingScalaVersions shouldBe "7.2.0 (Scala 3.0.0-RC3, 2.13)"
+  }
+
+  it(
+    "should list the Scala platform editions that support all cited versions of the Scala language"
+  ) {
+    ArtifactScalaVersionSupport(
+      Map(
+        `7.1.0` -> Seq(
+          ScalaNative(`3.0.0-M3`, Native.`0.3`),
+          ScalaNative(`2.13`, Native.`0.3`),
+          ScalaNative(`3.0.0-M3`, Native.`0.4`),
+          ScalaNative(`2.13`, Native.`0.4`)
+        )
+      )
+    ).summaryOfLatestArtifactsSupportingScalaVersions shouldBe "7.1.0 (Scala 3.0.0-M3, 2.13 - Native 0.4+0.3)"
```

Comment:
I don’t find this way of structuring things very clear.

BTW, what would be the output of:

~~~ scala
    ArtifactScalaVersionSupport(
      Map(
        `7.1.0` -> Seq(
          ScalaNative(`2.13`, Native.`0.3`),
          ScalaNative(`3.0.0-M3`, Native.`0.4`),
        )
      )
    ).summaryOfLatestArtifactsSupportingScalaVersions
~~~

Reply:
I've added a testcase in commit 3c23f5b to show what the result is: `7.1.0 (Scala 3.0.0-M3, 2.13)`. In this situation, there are a few options for what the badge text could be:

* Over promise: `7.1.0 (Scala 3.0.0-M3, 2.13 - Native 0.4+0.3)`, which incorrectly implies that there's a `3.0.0-M3` version for `Native 0.4`.
* Provide full detail: eg `7.1.0 (Scala 3.0.0-M3 - Native 0.4, Scala 2.13 - Native 0.3)`.
* Omit platform versions that don't have full support for the listed Scala language versions: `7.1.0 (Scala 3.0.0-M3, 2.13)`.

The endpoint `latest-by-scala-version.svg`, is intended to list all supported Scala versions, with the latest artifact version against them, and giving that information concisely for me takes priority over providing completely comprehensive information about the support for differing versions of the Scala platforms. It's also best not to over-promise, so of the three options I feel the last is the best, for this particular testcase.